src/core/ucc_global_opts.h

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCC_GLOBAL_OPTS_H_
+#define UCC_GLOBAL_OPTS_H_
+
+#include "config.h"
+
+#include <ucs/config/parser.h>
+#include <ucs/sys/preprocessor.h>
+#include <ucs/sys/compiler_def.h>
+#include <ucs/config/types.h>
+#include <ucs/debug/log_def.h>
+#include <ucs/config/parser.h>
+
+typedef struct ucc_global_config {
+    /* Log level above which log messages will be printed*/
+    ucs_log_component_config_t log_component;
+
+    /* Coll component libraries path */
+    char *component_path;
+    char *component_path_default;
+} ucc_global_config_t;
+
+extern ucc_global_config_t ucc_global_config;
+extern ucs_config_field_t  ucc_global_config_table[];
```

Comment:
Using UCS at some places directly and hiding/not used in some places (like malloc) make me uncomfortable. More and more, I think about this, we need a consistent approach. Should we create a UCS component under utilities and then use all UCS services underneath it? I know extra function indirection seems laborious and not ideal, but from a maintenance / future extensions perspective, we might be better off.

Reply:
So, seems we are back to where we started long ago. I remember we made a decision that we rely on UCS for all utilities. This is why we use it directly. For example, you are saying we want to abstract ucs_config_* set of APIs and then implement them using other components. There is no other existing implementation right now for this. Do you mean re-writing all this code ourselves inside ucc? Then why use services library in the first place? Malloc is another example - we have std "malloc" implementation already, but we don't have ucs_malloc exposed yet. I would use ucs_malloc directly if i had it already exposed in public ucs headers.

Lets discuss this in WG. But to me, abstracting UCS has little sense. Potential benefit is likely to never be realised. I mean, i don't see we will implement another services library internally.

Reply:
Update the PR as we discussed. Put all the ucs_ utilities under src/utils and re-defined them with ucc_ namespace.